### PR TITLE
Document start

### PR DESCRIPTION
**Pull request by faraplay**
Resolves issue #23.
Basically everything had to be made asynchronous so there is a lot of code change.

### CODE REVIEW

File: dev/my_style_loader/index.js
Comment (faraplay) [inline dev/my_style_loader/index.js:1]: `style-loader` takes ages to inject CSS for some reason, so I wrote this custom style loader

File: css/userSidebar.css [hunk 1]
```diff
@@ -31,9 +32,6 @@
     display: block;
     width: 80%;
 }
-#user-sidebar-open:before {
-    content: "menu";
-}
```
Comment (faraplay) [inline css/userSidebar.css:36]: Button content is now set in `addUserSidebarButton` in `header.js`

File: js/posts/index.js [hunk 2]
```diff
@@ -2,63 +2,62 @@ import setupNativeShare from "./nativeSharing.js";
...
     }
+    post.classList.add("riok");
+}
+
+export default function setupPosts() {
+    waitForAllElements(".sitetable.linklisting > .thing:not(.riok)", setupPost);
 }
+
 setupPosts();
-window.addEventListener("neverEndingLoad", setupPosts);
```
Comment (faraplay) [inline js/posts/index.js:64]: `waitForAllElements` will run the `setupPost` function on every new post that loads, so no need to listen for `neverEndingLoad`

File: js/sidebars.js
Comment (faraplay) [inline js/sidebars.js:1]: this was renamed to `sidebar/index.js`

File: js/subreddit_sidebar.js
Comment (faraplay) [inline js/subreddit_sidebar.js:1]: this was renamed to `sidebar/subredditSidebar.js`

**Comment by OctoNezd** [comment]
Looks great - had you tested it with user.js?

**Comment by faraplay** [comment]
Yes I've tested the tampermonkey script. I haven't tested it with Chrome/Kiwi though.
The userscript itself takes a bit of time to load in (because it's 4MB) but other than that it works. It might be worth making a new issue/feature request for splitting up the userscript into multiple files (especially the fonts).

**Pull request merged**